### ISSUE TITLE
Add a scan API similar to format

### ISSUE THREAD

**Issue by Viatorus** [Viatorus/emio#4]
Something like: 
```cpp
int i{};
emio::result<void> res = emio::scan("abc 123", "abc {}", i);
// i == 123
```

I have an old implementation of this but it didn't had compile-time checks and so but the structure of the code is already prepared for this.

**Comment by Viatorus** [comment]
Missing mandatory type: string

**Comment by Viatorus** [comment]
Fixed in #45.
